External/FEXCore/Source/Interface/Core/JIT/Arm64/MemoryOps.cpp

```diff
@@ -1340,6 +1340,165 @@ DEF_OP(StoreMemTSO) {
   }
 }
 
+DEF_OP(MemSet) {
+  const auto Op = IROp->C<IR::IROp_MemSet>();
+
+  const int32_t Size = Op->Size;
+  const auto MemReg = GetReg(Op->Addr.ID());
+  const auto Value = GetReg(Op->Value.ID());
+  const auto Length = GetReg(Op->Length.ID());
+  const auto Direction = GetReg(Op->Direction.ID());
+  const auto Dst = GetReg(Node);
+
+  // If Direction == 0 then:
+  //   MemReg is incremented (by size)
+  // else:
+  //   MemReg is decremented (by size)
+  //
+  // Counter is decremented regardless.
+
+  ARMEmitter::ForwardLabel BackwardImpl{};
+  ARMEmitter::ForwardLabel Done{};
+
+  mov(TMP1, Length.X());
+  if (Op->Prefix.IsInvalid()) {
+    mov(TMP2, MemReg.X());
+  }
+  else {
+    const auto Prefix = GetReg(Op->Prefix.ID());
+    add(TMP2, Prefix.X(), MemReg.X());
+  }
+
+  // Backward or forwards implementation depends on flag
+  cbnz(ARMEmitter::Size::i64Bit, Direction, &BackwardImpl);
+
+  auto MemStore = [&](auto OpSize, int32_t Size) {
```

Comment:
Using `auto` for lambda parameters turns them into templates, which aren't needed here (but might blow up compiler errors during future work). Just using `int`/`unsigned` would be better here.

Reply:
removed auto.